core/src/main/kotlin/de/fraunhofer/iem/kpiCalculator/core/strategy/RatioKPICalculationStrategy.kt

```diff
@@ -22,54 +23,49 @@ internal object RatioKPICalculationStrategy : BaseKpiCalculationStrategy() {
-    override fun internalCalculateKpi(
-        successScores: List<Pair<KpiCalculationResult.Success, Double>>,
-        failed: List<Pair<KpiCalculationResult, Double>>,
-        additionalWeight: Double,
-        hasIncompleteResults: Boolean,
-    ): KpiCalculationResult {
-
-        if (successScores.size != 2) {
+    override fun internalCalculateKpi(edges: Collection<KpiHierarchyEdge>): KpiCalculationResult {
+
+        if (edges.size != 2) {
             return KpiCalculationResult.Error(
                 "Ratio calculation strategy called " +
-                    "with ${successScores.size} valid elements, which is illegal."
+                    "with ${edges.size} valid elements, which is illegal."
             )
         }
 
-        val biggerValue =
-            if (successScores.first().first.score > successScores[1].first.score) {
-                successScores.first().first.score
-            } else {
-                successScores.last().first.score
-            }
+        if (edges.any { it.to.hasNoResult() }) {
+            return KpiCalculationResult.Error(
+                reason = "Ratio calculation strategy has elements without result"
+            )
+        }
 
-        val smallerValue =
-            if (successScores.first().first.score < successScores[1].first.score) {
-                successScores.first().first.score
-            } else {
-                successScores[1].first.score
-            }
+        val biggerValue: Double
+        val smallerValue: Double
 
-        if (biggerValue == smallerValue && smallerValue == 0) {
-            return KpiCalculationResult.Success(0)
+        val firstScore = edges.first().to.score * edges.first().actualWeight
+        val secondScore = edges.last().to.score * edges.last().actualWeight
+
+        if (firstScore > secondScore) {
+            biggerValue = firstScore
+            smallerValue = secondScore
+        } else {
+            biggerValue = secondScore
+            smallerValue = firstScore
         }
 
-        return try {
-            if (!hasIncompleteResults) {
-                KpiCalculationResult.Success(
-                    score = ((smallerValue.toDouble() / biggerValue.toDouble()) * 100).toInt()
-                )
-            } else {
-                KpiCalculationResult.Incomplete(
-                    score = ((smallerValue.toDouble() / biggerValue.toDouble()) * 100).toInt(),
-                    reason = "Incomplete results.",
-                    additionalWeights = 0.0,
-                )
+        val ratio =
+            try {
+                if (biggerValue != 0.0) {
+                    smallerValue / biggerValue
+                } else {
+                    // NB: for whatever reason the statement above results in -Infinity
+                    // instead of an ArithmeticException when  dividing by 0.
+                    throw ArithmeticException("Tried division by 0")
```

Comment:
also, throwing and immediately re-catching seems a little odd. At some day I learned, exceptions should not be (ab)used to handle program flow ;)